docs/ABI/Mangling.rst

```diff
@@ -1041,3 +1041,46 @@ Some kinds need arguments, which precede ``Tf``.
 If the first character of the string literal is a digit ``[0-9]`` or an
 underscore ``_``, the identifier for the string literal is prefixed with an
 additional underscore ``_``.
+
+Conventions for foreign symbols
+-------------------------------
+
+Swift interoperates with multiple other languages - C, C++, Objective-C, and
+Objective-C++. Each of these languages defines their own mangling conventions,
+so Swift must take care to follow them. However, these conventions do not
+restrict Swift-specific constructs like Swift type metadata, so the Swift
+language may impose its own conventions in these cases.
+
+Importing C and C++ structs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We don't touch the mangling of the original symbol. Since Swift metadata symbols
+and value-witness-tables are not generated by the C/C++ compiler, Swift generates
+them when C/C++ module is imported and stores them in the ``__C`` module. This can
```

Comment:
```suggestion
Types imported from C and C++ are imported as if they are located in the ``__C`` module, regardless of the actual Clang module that they are coming from. This can be observed ...

<add the mangling example here>

The C/C++ compiler does not generate Swift metadata symbols and value witness tables for C and C++ types. To make a foreign type usable in Swift in the same way as a native type, the Swift compiler must generate these symbols. Specifically, each Swift module that uses a given C or C++ type generates the necessary Swift symbols.

<show the symbols here>

```